modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/RefreshTokenHandler.java

```diff
@@ -0,0 +1,153 @@
+import org.wso2.mdm.agent.proxy.utils.ServerUtilities;
+import org.wso2.mdm.agent.proxy.utils.Constants.HTTP_METHODS;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Getting new access token and refresh token after access token expiration
+ */
+public class RefreshTokenHandler {
+	private static final String TAG = "RefreshTokenHandler";
+	private static final String SCOPE_LABEL = "scope";
+	private static final String PRODUCTION_LABEL = "PRODUCTION";
+	private static final DateFormat dateFormat =
+			new SimpleDateFormat("MM/dd/yyyy HH:mm:ss", Locale.getDefault());
+	private static final String COLON = ":";
+	private Token token;
+
+	public RefreshTokenHandler(Token token) {
+		this.token = token;
+	}
+
+	public void obtainNewAccessToken() {
+		new NetworkCallTask().execute();
+	}
+
+	private class NetworkCallTask extends AsyncTask<String, Void, String> {
+
+		private String responseCode = null;
+
+		@Override
+		protected String doInBackground(String... params) {
+			String response = null;
+
+			Map<String, String> requestParams = new HashMap<String, String>();
+			requestParams.put(Constants.GRANT_TYPE, Constants.REFRESH_TOKEN);
+			requestParams.put(Constants.REFRESH_TOKEN, token.getRefreshToken());
+			requestParams.put(SCOPE_LABEL, PRODUCTION_LABEL);
+			EndPointInfo apiUtilities = new EndPointInfo();
+			apiUtilities.setEndPoint(IdentityProxy.getInstance().getAccessTokenURL());
+			apiUtilities.setHttpMethod(HTTP_METHODS.POST);
+			apiUtilities.setRequestParamsMap(requestParams);
+
+			byte[] credentials = Base64.encodeBase64((IdentityProxy.clientID + COLON +
+			                                          IdentityProxy.clientSecret).getBytes());
+			String encodedCredentials = new String(credentials);
+
+			Map<String, String> headers = new HashMap<String, String>();
+
+			String authorizationString = Constants.AUTHORIZATION_MODE + encodedCredentials;
+			headers.put(Constants.AUTHORIZATION_HEADER, authorizationString);
+			headers.put(Constants.CONTENT_TYPE_HEADER, Constants.DEFAULT_CONTENT_TYPE);
+
+			Map<String, String> responseParams = null;
+			try {
+				responseParams = ServerUtilities.postDataAPI(apiUtilities, headers);
+			} catch (IDPTokenManagerException e) {
+				Log.e(TAG, "Failed to contact server." + e);
+			}
+
+			response = responseParams.get(Constants.SERVER_RESPONSE_BODY);
+			responseCode = responseParams.get(Constants.SERVER_RESPONSE_STATUS);
+			return response;
+		}
+
+		@SuppressLint("SimpleDateFormat")
+		@Override
+		protected void onPostExecute(String result) {
+
+			String refreshToken = null;
+			String accessToken = null;
+			int timeToExpireSecond = 3000;
+			IdentityProxy identityProxy = IdentityProxy.getInstance();
+			try {
+				JSONObject response = new JSONObject(result);
+
+				if (responseCode != null && responseCode.equals(Constants.REQUEST_SUCCESSFUL)) {
```

Comment:
If the value assigned to "responseCode" is compared against Constants.REQUEST_SUCCESSFUL, you could have avoided that redundant check to see if responsCode is null.